src/Core/Geometry/TopologicalMesh.cpp

```diff
@@ -433,6 +594,332 @@ bool TopologicalMesh::splitEdge( TopologicalMesh::EdgeHandle eh, Scalar f ) {
+    auto wd0       = m_wedges.getWedgeData( v0widx );
+    auto v1widx    = property( m_wedgeIndexPph, a0 );
+    auto wd1       = m_wedges.getWedgeData( v1widx );
+    auto wd        = interpolateWedgeAttributes( wd0, wd1, f );
+    wd.m_position  = p;
+    auto widx      = m_wedges.add( wd );
+
+    base::split_edge( eh, v );
+
+    auto r0 = find_halfedge( v1, v );
+    auto r1 = next_halfedge_handle( r0 );
+    auto r2 = next_halfedge_handle( r1 );
+
+    auto s0 = find_halfedge( v, v0 );
+    auto s1 = next_halfedge_handle( s0 );
+    auto s2 = next_halfedge_handle( s1 );
+
+    auto t0 = find_halfedge( v0, v );
+    auto t1 = next_halfedge_handle( t0 );
+    auto t2 = next_halfedge_handle( t1 );
+
+    auto u0 = find_halfedge( v, v1 );
+    auto u1 = next_halfedge_handle( u0 );
+    auto u2 = next_halfedge_handle( u1 );
+
+    auto updateWedgeIndex1 = [this, widx]( HalfedgeHandle t0_,
+                                           HalfedgeHandle t1_,
+                                           HalfedgeHandle t2_,
+                                           HalfedgeHandle a1_,
+                                           HalfedgeHandle a2_ ) {
+        if ( !is_boundary( t0_ ) )
+        {
+            CORE_ASSERT( t2_ == a2_, "TopologicalMesh: splitEdgeWedge inconsistency" );
+            property( this->m_wedgeIndexPph, t0_ ) = widx;
+            property( this->m_wedgeIndexPph, t1_ ) =
+                this->m_wedges.newReference( property( this->m_wedgeIndexPph, a1_ ) );
+        }
+        else
+        { property( this->m_wedgeIndexPph, t0_ ) = WedgeIndex(); }
+    };
+
+    updateWedgeIndex1( t0, t1, t2, a1, a2 );
+    updateWedgeIndex1( r0, r1, r2, b1, b2 );
+
+    auto updateWedgeIndex2 = [this, widx]( HalfedgeHandle s0_,
+                                           HalfedgeHandle s1_,
+                                           HalfedgeHandle s2_,
+                                           HalfedgeHandle b1_,
+                                           TopologicalMesh::WedgeIndex v0widx_ ) {
+        if ( !is_boundary( s0_ ) )
+        {
+            CORE_ASSERT( s1_ == b1_, "TopologicalMesh: splitEdgeWedge inconsistency" );
+            property( this->m_wedgeIndexPph, s2_ ) = widx;
+            property( this->m_wedgeIndexPph, s0_ ) = this->m_wedges.newReference( v0widx_ );
+        }
+        else
+        { property( this->m_wedgeIndexPph, s0_ ) = WedgeIndex(); }
+    };
+
+    updateWedgeIndex2( s0, s1, s2, b1, v0widx );
+    updateWedgeIndex2( u0, u1, u2, a1, v1widx );
+
+    return true;
+}
+
+void TopologicalMesh::collapseWedge( TopologicalMesh::HalfedgeHandle heh ) {
+
+    HalfedgeHandle h  = heh;
+    HalfedgeHandle hn = next_halfedge_handle( h );
+    HalfedgeHandle hp = prev_halfedge_handle( h );
+
+    HalfedgeHandle o  = opposite_halfedge_handle( h );
+    HalfedgeHandle on = next_halfedge_handle( o );
+    HalfedgeHandle op = prev_halfedge_handle( o );
+
+    FaceHandle fh = face_handle( h );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'fh' is assigned a value that is never used.](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=4930613)